Add implicit branch creation and other small changes to Readme.md (#48)
appliedAI-Initiative/lakefs-spec: README.md

df = pd.read_parquet('lakefs://quickstart/main/lakes.parquet', storage_options=storage_options)
```

You can then update data in LakeFS like so:

```python
df.to_csv('lakefs://quickstart/main/lakes.parquet', storage_options=storage_options)
```

If the target file does not exists, it is created, therwise, the existing file is updated.

If the specified branch does not exist, it is created by default. This behaviour can be set in the filesystem constructor via the `create_branch_ok` flag.

```python
from lakefs_spec import LakeFSFileSystem

> # The default setting, precheck_files=True enables implicit branch creation

Reviewer: ```suggestion
# create_branch_ok=True (the default setting) enables implicit branch creation
```

URIs need to have the form `lakefs://<repo>/<ref>/<resource>`, with the repository name, the ref name (either a branch name or a commit SHA, depending on the operation), and resource name.
The resource can be a single file name, or a directory name for recursive operations.

### Updating files

You can update data in LakeFS like so:

```python
df.to_csv('lakefs://quickstart/main/lakes.parquet', storage_options=storage_options)
```

> When the target file not exists, it is created. Otherwise, the existing file is updated.

Reviewer: ```suggestion
If the target file does not exist, it is created, otherwise, the existing file is updated.
```

> If the specified branch does not exist, it is created. This behaviour can be set in the filesystem constructor via the `create_branch_ok` flag:

Reviewer: ```suggestion
If the specified branch does not exist, it is created by default. This behaviour can be set in the filesystem constructor via the `create_branch_ok` flag:
```

```python
from lakefs_spec import LakeFSFileSystem

> # The default setting, precheck_files=False disables client-side caching.

Reviewer: This comment is stale / copy-pasted?

Author: jup

        message = f"❌ Remove file {ctx.resource}"
    else:
        message = f"✅ Add file {ctx.resource}"

>     

Reviewer: Please revert, the space makes it more readable because of the different indentations (imo).

### Scoped filesystem behavior changes

> To selectively enable or disable automatic commits, client-side caching, or implicit branch creation you can use a `scope` context manager:

Reviewer: ```suggestion
To selectively enable or disable automatic commits, client-side caching, or automatic branch creation, you can use a `scope` context manager:
```
